### PR TITLE
M3 IMPL: PR Number 3 - DTO: Request and response

### PR DESCRIPTION
**Pull request by Sidarthapati**
# Pull Request
  
  ## Summary
  Implements Phase 4 of M3: 14 Java record DTOs for the grid module, split into
  `dto/request/` and `dto/response/` subpackages with clarified naming for all
  proposal and override types.
  
  ---
  
  ## What Changed
  - Added `dto/request/` with 4 records: `RegionEditRequest`, `ProposalRejectRequest`,
    `IntradayReassignmentRequest`, `TileShareRequest`
  - Added `dto/response/` with 10 records: `ProposalResponse`, `RegionResponse`,
    `AssignmentResponse`, `DaAssignmentResponse`, `IntradayReassignmentResponse`,
    `TileShareResponse`, `GridVertexResponse`, `ServiceabilityResponse`,
    `TileAtResponse`, `TileLoadScoreResponse`
  - Renamed `OverrideRequest` → `RegionEditRequest` to distinguish Scenario A
    (pre-approval region edit) from `INTRADAY_OVERRIDE` proposals (Scenario B)
  - Renamed `ProposalDto` → `ProposalResponse`, `RegionDto` → `RegionResponse`
    for naming consistency
  - Added `INTRADAY_SHARE` to `ProposalType` enum (required for tile-share proposal
    lifecycle — creates a distinct proposal type from `INTRADAY_OVERRIDE`) 
  - Updated `M3-GRID-DESIGN.md` and `M3-IMPLEMENTATION-PLAN.md` to reflect the
    two intraday assignment patterns (override vs. tile share)
    
  ---
  
  ## Why This Change?
  Phase 5 service interfaces (`ProposalService`, `GridService`, etc.) return and
  accept these DTOs. Establishing request/response separation now prevents
  rename churn once services are wired to controllers in Phase 8. The
  `RegionEditRequest` rename eliminates a naming collision — "override" was being
  used for both Scenario A (in-place edit of a not-yet-approved proposal) and
  Scenario B (new `INTRADAY_OVERRIDE` proposal on an active plan), which are
  meaningfully different operations.
  
  ---
  
  ## Testing
  - [x] Tested locally
  - [x] Edge cases considered
  - [x] No breaking changes introduced
  
  ### Test Evidence
  $ mvn clean install -pl common,grid -am -q
  BUILD SUCCESS
  No compilation errors. Phase 5 service layer not yet started so no downstream
  breakage possible.
  
  ---
  
  ## Impact
  
  ### Areas Affected
  - [x] Backend
  - [ ] Frontend
  - [ ] Routing
  - [ ] Infra
  - [ ] Database
  - [ ] NA
  
  ### Modules Affected
  - [ ] M1 — Auth
  - [ ] M2 — Pricing
  - [x] M3 — Grid
  - [ ] M4 — Orders
  - [ ] M5 — Dispatch
  - [ ] M6 — Routing
  - [ ] M7 — Hub
  - [ ] M8 — Barcode
  - [ ] M9 — Airline
  - [ ] M10 — SLA
  - [ ] M11 — Exceptions
  - [ ] NA
  
  ---
  
  ## Risks / Concerns
  - None. Phase 5 (service layer) hasn't started so there are no callers of these
    DTOs yet. The rename from `OverrideRequest` to `RegionEditRequest` carries zero
    breakage risk at this stage.
    
  ---
  
  ## Checklist
  - [x] Code follows project conventions (package layout, no cross-module internal imports)
  - [x] Self-review completed
  - [x] Append-only audit trail preserved (no mutations to scan/manifest/role-change records)
  - [x] No sensitive data or credentials exposed
  - [x] Documentation updated if behaviour changed
  - [x] Ready for review

### CODE REVIEW

File: grid/src/main/java/com/oneday/grid/dto/request/IntradayReassignmentRequest.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.List;
+import java.util.UUID;
+
+// Scenario B only: tile move on an already-ACTIVE plan.
+// Creates a brand-new INTRADAY_OVERRIDE proposal that requires its own approval.
+public record IntradayReassignmentRequest(
+        UUID cityId,
+        UUID fromDaId,
+        UUID toDaId,
+        List<UUID> tileIdsToMove,
```
Comment (BanerjeeAgniva) [inline grid/src/main/java/com/oneday/grid/dto/request/IntradayReassignmentRequest.java:12]: whats this? please give example where it gets populated and what function it serves?

Comment (Sidarthapati) [reply grid/src/main/java/com/oneday/grid/dto/request/IntradayReassignmentRequest.java:12]: Example:

  DA-7 is overloaded mid-shift — tiles [T3, T4] in their territory are swamped with pickups. The station manager opens the dashboard and drags T3 and T4 onto DA-12's zone. That gesture fires:

  {
    "cityId": "...",
    "fromDaId": "da-7-uuid",
    "toDaId":   "da-12-uuid",
    "tileIdsToMove": ["t3-uuid", "t4-uuid"],
    "requestedBy": "station-mgr-uuid"
  }

  What the service does with it:

  1. Looks up the current ACTIVE DaTileAssignment rows for fromDaId and strips out the listed tiles → DA-7's new territory.
  2. Adds those tiles to toDaId's territory.
  3. Runs contiguity checks on both resulting territories (a DA can't end up with a fragmented, disconnected zone).
  4. Checks that toDaId's load after absorbing the tiles doesn't breach daMaxLoad.
  5. If valid, persists a new AssignmentProposal of type INTRADAY_OVERRIDE with status PROPOSED — it still needs station manager approval before going ACTIVE, which is why requestedBy is captured (audit trail
   for who triggered it).

  The reason it's a list and not a single tile: a pincode area might span two adjacent tiles, and moving just one would leave the other orphaned on DA-7 but unreachable as a contiguous zone. The manager picks
   the logical unit to move, which can be 1–N tiles.

File: grid/src/main/java/com/oneday/grid/dto/response/TileShareResponse.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.oneday.grid.domain.ProposalStatus;
+
+import java.time.Instant;
+import java.util.UUID;
+
+public record TileShareResponse(
+        UUID proposalId,
+        UUID daId,
+        UUID tileId,
+        ProposalStatus status,
```
Comment (BanerjeeAgniva) [inline grid/src/main/java/com/oneday/grid/dto/response/TileShareResponse.java:12]: The tile share response approved by whom? where is that being captured? is it needed ? will it be there in status itself?

File: grid/src/main/java/com/oneday/grid/dto/response/TileLoadScoreResponse.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import java.time.LocalDate;
+import java.util.UUID;
+
+public record TileLoadScoreResponse(
+        UUID tileId,
+        LocalDate date,
+        int unservedOrders,
+        double adjustedLoadScore,
+        String severity
```
Comment (BanerjeeAgniva) [inline grid/src/main/java/com/oneday/grid/dto/response/TileLoadScoreResponse.java:11]: what info will be stored in this variable?

File: grid/src/main/java/com/oneday/grid/dto/response/TileLoadScoreResponse.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.oneday.grid.dto.response;
+
+import java.time.LocalDate;
+import java.util.UUID;
+
+public record TileLoadScoreResponse(
+        UUID tileId,
+        LocalDate date,
+        int unservedOrders,
+        double adjustedLoadScore,
```
Comment (BanerjeeAgniva) [inline grid/src/main/java/com/oneday/grid/dto/response/TileLoadScoreResponse.java:10]: how are we calculating this --> if link to any documentation/  or previously missed code / or will it come in the upcoming PRs for service layer?

File: grid/src/main/java/com/oneday/grid/dto/response/RegionResponse.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.oneday.grid.dto.response;
+
+import java.util.List;
+import java.util.UUID;
+
+public record RegionResponse(
+        UUID id,
+        UUID daId,
+        int nDasRequired,
+        double estimatedDemandMin,
```
Comment (BanerjeeAgniva) [inline grid/src/main/java/com/oneday/grid/dto/response/RegionResponse.java:10]: again same question , what is the source of these values, will they come up later in the service layer implementation?
estimatedDemandMin
estimatedUtilPct
hasBootstrappedTiles

File: grid/src/main/java/com/oneday/grid/dto/response/ProposalResponse.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        Double optimalityGapPct,
+        int totalDas,
+        Double coveragePct,
+        List<UUID> understaffedTileIds,
+        Instant proposedAt,
+        UUID reviewedBy,
+        Instant reviewedAt,
+        String notes,
+        List<RegionResponse> regions
+) {}
```
Comment (BanerjeeAgniva) [inline grid/src/main/java/com/oneday/grid/dto/response/ProposalResponse.java:30]: on a high level, who is rending a proposal request and who is consuming the reponse / from a module perspective , just for High level understanding

File: grid/src/main/java/com/oneday/grid/dto/request/TileShareRequest.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.oneday.grid.dto.request;
+
+import java.util.UUID;
+
+public record TileShareRequest(
+        UUID cityId,
+        UUID daId,
+        UUID tileId,
+        UUID requestedBy
+) {}
```
Comment (BanerjeeAgniva) [inline grid/src/main/java/com/oneday/grid/dto/request/TileShareRequest.java:10]: who will be making such a request in day to day operations from a high level perspective?

File: grid/src/main/java/com/oneday/grid/domain/ProposalType.java [hunk 8]
```diff
@@ -1,5 +1,5 @@
 package com.oneday.grid.domain;
 
 public enum ProposalType {
-    NIGHTLY, INTRADAY_OVERRIDE, INTRADAY_SUGGESTION
+    NIGHTLY, INTRADAY_OVERRIDE, INTRADAY_SUGGESTION, INTRADAY_SHARE
```
Comment (BanerjeeAgniva) [inline grid/src/main/java/com/oneday/grid/domain/ProposalType.java:4]: i guess most of our proposals will come nightly , for the nightly cron job that we will set up as discuss , can you discuss example scnearios of the other types

File: grid/src/main/java/com/oneday/grid/dto/request/TileShareRequest.java (hunk 7 above)
Comment (Sidarthapati) [reply grid/src/main/java/com/oneday/grid/dto/request/TileShareRequest.java:10]: It would be a station manager or supervisor. They would be monitoring current statuses, hence the intraday override options

File: grid/src/main/java/com/oneday/grid/domain/ProposalType.java (hunk 8 above)
Comment (Sidarthapati) [reply grid/src/main/java/com/oneday/grid/domain/ProposalType.java:4]: INTRADAY_OVERRIDE 

  The station manager manually intervenes on an already-active plan. System doesn't initiate this — a human does.

  Example: DA-7 calls in sick at 10 AM. Their 6 tiles need to be redistributed right now. The manager opens the dashboard, drags T3+T4 onto DA-12 and T5+T6 onto DA-9, and submits. This creates an
  INTRADAY_OVERRIDE proposal. It still needs a second manager to approve (four-eyes), then it goes ACTIVE and supersedes the nightly plan for those tiles. Produces new DaTileAssignment rows for the affected
  tiles.
  
  ---
  INTRADAY_SUGGESTION
  
  System-generated, not manager-generated. The IntradayMonitorJob (Phase 6.4) watches real-time queue depth per tile. If tile T8 has been at >1.5× expected load for 15+ sustained minutes, the system
  automatically creates an INTRADAY_SUGGESTION proposal — "consider moving T8 from DA-3 to DA-11." The manager sees it as a flagged alert, not a mandate. They can approve, reject, or dismiss. No one typed a
  request — the algorithm noticed the overload and raised its hand.

  The distinction from OVERRIDE: origin. SUGGESTION = algorithm noticed something. OVERRIDE = manager decided something.

  ---
  INTRADAY_SHARE
  
  Instead of moving a tile fully from DA-A to DA-B, both DAs work the same tile simultaneously for a defined time window. Used when a tile has a sudden demand spike too large for one DA but too small to
  justify a full territory reshuffle.

  Example: A 200-order B2B drop arrives at tile T2 at noon (a corporate bulk pickup). DA-5 who owns T2 can't absorb it alone within shift. Rather than reassigning T2 permanently, the manager submits an
  INTRADAY_SHARE: "DA-5 and DA-14 both cover T2 for the next 3 hours." The nDasOnTile column on DaTileAssignment goes from 1 → 2 for that window. After the window, DA-14 returns to their normal territory. No
  territory boundary is permanently redrawn.

File: grid/src/main/java/com/oneday/grid/dto/response/ProposalResponse.java (hunk 6 above)
Comment (Sidarthapati) [reply grid/src/main/java/com/oneday/grid/dto/response/ProposalResponse.java:30]: Night job is proposed by - System, approved by station manager
Intraday monitoring proposed by- System , approved by station manager
Intraday Overrides (Both tileshare and tile reassign) proposed by - Station manager, approved by another station manager, supervisor

File: grid/src/main/java/com/oneday/grid/dto/response/RegionResponse.java (hunk 5 above)
Comment (Sidarthapati) [reply grid/src/main/java/com/oneday/grid/dto/response/RegionResponse.java:10]: Yes, these would be received from M4 dbs Yes they will come up in service layers too, but source of these values lies in m4 order histories.

File: grid/src/main/java/com/oneday/grid/dto/response/TileLoadScoreResponse.java (hunk 4 above)
Comment (Sidarthapati) [reply grid/src/main/java/com/oneday/grid/dto/response/TileLoadScoreResponse.java:10]: Theres no missed code , all impl will come up in service layers. This load score would be received from m5.(from queue depth)

File: grid/src/main/java/com/oneday/grid/dto/response/TileLoadScoreResponse.java (hunk 3 above)
Comment (Sidarthapati) [reply grid/src/main/java/com/oneday/grid/dto/response/TileLoadScoreResponse.java:11]: This would be the current state of tile, in case of higher severity action needs to be taken. Intraday monitoring will do this .

File: grid/src/main/java/com/oneday/grid/dto/response/TileShareResponse.java (hunk 2 above)
Comment (Sidarthapati) [reply grid/src/main/java/com/oneday/grid/dto/response/TileShareResponse.java:12]: Approved by another station manager in same city or a supervisor of that city.

**Review by BanerjeeAgniva — APPROVED** [review]
LGTM
